policy-engine/src/graph.rs

```diff
@@ -61,14 +61,14 @@ pub(crate) fn index(req: HttpRequest) -> Box<Future<Item = HttpResponse, Error =
     if let Err(e) = commons::ensure_query_params(mandatory_params, req.query_string()) {
         return Box::new(future::err(e));
     }
-    // TODO(steveeJ): take another look at the actix-web docs for a method that
-    // provides this parameters split.
```

Comment:
Have you seen/considered https://docs.rs/actix-web/1.0.5/actix_web/web/struct.Query.html? Can this be done by deserializing to a serde-transparent `HashMap<String, String>`?

Reply:
I saw it but I couldn't make much sense of it as the docstring seems to be outdated. It mentions a `username` field which doesn't exist in the example code. Further it's not obvious to me how to access the request headers if the `Query` type is used.

Reply:
Looking at the docs again I saw `Query::from_query` which works for our needs :raised_hands: I added a commit on top which I'll squash if we agree on it.